src/pybind/mgr/dashboard/services/ceph_service.py

```diff
@@ -173,6 +173,44 @@ def send_command(cls, srv_type, prefix, srv_spec='', **kwargs):
         except Exception:  # pylint: disable=broad-except
             return outb
 
+    @staticmethod
+    def get_smart_data_by_device(device):
+        dev_id = device['devid']
+        if 'daemons' in device and device['daemons']:
+            svc_type, svc_id = [daemon for daemon in device['daemons']
+                                if daemon.startswith('mon')][0].split('.')
```

Comment:
```suggestion
                                if daemon.startswith('osd')][0].split('.')
```
Should we use `osd` here? I got this error:

```
  File "/usr/share/ceph/mgr/dashboard/controllers/__init__.py", line 657, in inner
    ret = func(*args, **kwargs)
  File "/usr/share/ceph/mgr/dashboard/controllers/__init__.py", line 850, in wrapper
    return func(*vpath, **params)
  File "/usr/share/ceph/mgr/dashboard/controllers/host.py", line 115, in smart
    return CephService.get_smart_data_by_host(hostname)
  File "/usr/share/ceph/mgr/dashboard/services/ceph_service.py", line 202, in get_smart_data_by_host
    for device in CephService.get_devices_by_host(hostname)
  File "/usr/share/ceph/mgr/dashboard/services/ceph_service.py", line 182, in get_smart_data_by_device
    if daemon.startswith('mon')][0].split('.')
IndexError: list index out of range
```

And my `device` variable in this call:
```
{u'daemons': [u'osd.0'], u'devid': u'node1-disk3', u'location': [{u'host'
: u'node1', u'dev': u'vdc'}]}
```

I'm not sure if a device can has a `mon` daemon with it. I'm testing this PR in a Rook-spawn cluster.
Also, would it be nice if we can check if the list is empty before getting it's first element?

BTW the [helper fixture](https://gist.githubusercontent.com/p-se/84cc04b0f18e90c25eb1bebbf92c8bc8/raw/130e2305169a912a74c8e4ebb232bba25d3c19a8/host.patch) in PR description can't be applied directly because `get_smart_data_by_hostname` is renamed to `get_smart_data_by_host`.

Reply:
You're absolutely right. I adapted the code to use 'osd' instead of 'mon'. It checks now if the list is empty before it tries to access the first element and I fixed the patch to mock the data.